File: src/main/java/ru/highloadcup/check/VisitEOCustomChecker.java
package ru.highloadcup.check;

import org.springframework.stereotype.Component;
import ru.highloadcup.api.VisitEO;

@Component
public class VisitEOCustomChecker implements CustomChecker<VisitEO> {
    @Override
    public void checkFullyNull(VisitEO object) {
        if (object.getLocationId() == null
                && object.getUserId() == null
                && object.getVisitedAt() == null
                && object.getMark() == null) {
            throw new IllegalArgumentException("object is fully null");
        }
    }
}

File: src/main/java/ru/highloadcup/check/VisitEOCustomChecker.kt
package ru.highloadcup.check

import org.springframework.stereotype.Component
import ru.highloadcup.AllOpen
import ru.highloadcup.api.VisitEO

@AllOpen
@Component
class VisitEOCustomChecker : CustomChecker<VisitEO> {
    override fun checkFullyNull(checkingObject: VisitEO) {
        if (checkingObject.locationId == null
                && checkingObject.userId == null
                && checkingObject.visitedAt == null
                && checkingObject.mark == null) {
            throw IllegalArgumentException("checkingObject is fully null")
        }
    }
}
